src/compile/selection/transforms/project.ts

```diff
@@ -72,12 +73,23 @@ const project: TransformCompiler = {
       }
     }
 
-    if (init) {
+    if (selDef.init) {
       if (scales.has(selCmpt)) {
         log.warn(log.message.NO_INIT_SCALE_BINDINGS);
       } else {
-        const parseInit = (i: any) => proj.map(p => (i[p.channel] !== undefined ? i[p.channel] : i[p.field]));
-        selCmpt.init = selCmpt.type === 'interval' ? parseInit(init) : array(init).map(parseInit);
+        function parseInit<T extends SelectionInitMapping | SelectionInitArrayMapping>(i: T): T['a'][] {
+          return proj.map(p => (i[p.channel] !== undefined ? i[p.channel] : i[p.field]));
+        }
+
+        if (isIntervalSelection(selDef)) {
+          const initCmpt = parseInit(selDef.init);
+          selCmpt.init = initCmpt;
+        } else {
+          const init = isArray(selDef.init) ? selDef.init : [selDef.init];
+          const initCmpt = init.map(parseInit);
+
+          selCmpt.init = initCmpt;
```

Comment:
@arvind I tried to modify the code on the left to have more strict typings, and it seems like the logic is incorrect as it does not produce a matching type.  

For this `else` case, the initCmpt is `SelectionInit[][]`, but `init` of a selection component expects `(SelectionInit | SelectionInitArray)[]`.  So I wonder if you forget to flatten here.

Reply:
NVM, I think I got the logic now. Let me fix and push.